CMakeLists.txt

```diff
@@ -145,6 +145,23 @@ if(CLIENTS OR TESTS)
   endif()
 endif()
 
+# Check plugin support
+find_path(DLFCN_H "dlfcn.h")
+if(DLFCN_H)
+  message(STATUS "Found <dlfcn.h>, enabling plugins using POSIX dynamic linker interface")
+  set(PLUGINS TRUE)
+  set(SOURCES_PLUGINS
+    src/utilities/plugin_posix.cpp
+  )
+endif()
+
+if(NOT PLUGINS)
+  message(STATUS "Disabling plugin support")
+  set(SOURCES_PLUGINS
+    src/utilities/plugin_none.cpp
+  )
+endif()
+
```

Comment:
Just an idea: would it be useful to guard all the Plugin-related things with a CMake option (e.g. similar to the current SAMPLES or CLIENTS options)? Not sure if the remainder of the code is now dependent on the plugin-system, but if it isn't, this might be a nice small addition. Most users won't need the plugin-system I guess, and this way it might be easier to support older compilers. It also saves another installed header.